internal/tlsprofile/get.go

```diff
@@ -0,0 +1,45 @@
+package tlsprofile
+
+import (
+	"context"
+	"errors"
+
+	"github.com/go-logr/logr"
+	openshiftconfigv1 "github.com/openshift/api/config/v1"
+
+	configv1alpha1 "github.com/os-observability/tempo-operator/apis/config/v1alpha1"
+)
+
+// ErrGetProfileFromCluster return when failed to get the cluster security policy in openshift.
+var ErrGetProfileFromCluster = errors.New("failed to get profile from cluster, using default TLS profile")
+
+// Get the profile according to the features configuration, if the policy is invalid or is not specified (empty string) this
+// should return the default profile (Intermediate), if openshift.ClusterTLSPolicy is enabled, it should get the profile
+// from the cluster, if the cluster return a unknow profile this should return an error.
+func Get(ctx context.Context, fg configv1alpha1.FeatureGates, c k8getter, log logr.Logger) (TLSProfileOptions, error) {
+	var tlsProfileType openshiftconfigv1.TLSSecurityProfile
+	var err error
+	var returnedErr error
+	// If ClusterTLSPolicy is enabled get the policy from the cluster
+	if fg.OpenShift.ClusterTLSPolicy {
+		tlsProfileType, err = getTLSProfileFromCluster(ctx, c)
+		if err != nil {
+			returnedErr = ErrGetProfileFromCluster
```

Comment:
Could we return here immediately with `return TLSProfileOptions{}, ErrGetProfileFromCluster`?

Reply:
If we do that we won't be able to return the default tls options.

Reply:
Maybe I'm misunderstanding something, but afaics the only call to `tlsprofile.Get` is in `microservices_controller.go`, and if this function returns an `ErrGetProfileFromCluster` error than the `tlsProfile` variable is unused?